src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -98,33 +98,30 @@ public void Deconstruct(out int a, out string b)
             comp.VerifyDiagnostics();
             comp.VerifyIL("C.Main", @"
 {
-  // Code size       48 (0x30)
+  // Code size       44 (0x2c)
   .maxstack  3
   .locals init (long V_0, //x
                 string V_1, //y
                 int V_2,
-                string V_3,
-                string V_4)
+                string V_3)
   IL_0000:  newobj     ""C..ctor()""
   IL_0005:  ldloca.s   V_2
   IL_0007:  ldloca.s   V_3
-  IL_0009:  call       ""void C.Deconstruct(out int, out string)""
+  IL_0009:  callvirt   ""void C.Deconstruct(out int, out string)""
```

Comment:
that is ok. Most likely reason is that Deconstruct is now called off a temporary.

I.E. you had

```C#
(new C1()).Deconstruct
```

now it is in, probably more conservatively correct, form

```C#
var ephemeralTemp = new C();
ephemeralTemp.Deconstruct
```

Optimizer kills ephemeralTemp, but the call still looks like applied to something that could be `null`

Not a big deal. If needed to be fixed, it should probably be fixed in the IL generator.


---
In reply to: [100636485](https://github.com/dotnet/roslyn/pull/17027#discussion_r100636485) [](ancestors = 100636485)